src/Google/RequestReviewStatuses.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Google;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+
+/**
+ * Helper class for Account request Review feature
+ */
+class RequestReviewStatuses implements Service {
+
+	public const DISAPPROVED    = 'DISAPPROVED';
+	public const WARNING        = 'WARNING';
+	public const ELIGIBLE       = 'ELIGIBLE';
+	public const UNDER_REVIEW   = 'UNDER_REVIEW';
+	public const PENDING_REVIEW = 'PENDING_REVIEW';
+	public const ONBOARDING     = 'ONBOARDING';
+
+	/**
+	 * Merges the different program statuses based on priority
+	 *
+	 * @param array $response
+	 *
+	 * @return array
+	 */
+	public function get_statuses_from_response( array $response ) {
+		$issues         = [];
+		$review_allowed = false;
+		$status         = 'APPROVED';
+
+		foreach ( $response as $program_type ) {
+			foreach ( $program_type['data']['regionStatuses'] as $region_status ) {
+
+				if ( $region_status['reviewEligibilityStatus'] === $this::ELIGIBLE ) {
```

Comment:
Just a quick comment, shouldn't the `constant` - `$this::ELIGIBLE` be invoked with `self` instead of `$this`? The main reason is that usually `$this` refers to the current object and in this case, we are referring to the current class.

Reply:
Thanks for the Catch @jorgemd24 

You're right. PHP self refers to the class members, but not for any particular object. 

Fixed here: https://github.com/woocommerce/google-listings-and-ads/pull/1413/commits/4310f5e545790ba43ed19d74b18d4e77edd8c7b6